### PR TITLE
example: Modify `taproot-psbt.rs` to make the use of prevouts clearer.

### PR DESCRIPTION
**Pull request by ssantos21**
The `taroot-psbt.rs` example uses only one input, and therefore the current code may not make it clear that the number of prevout items must correspond to the number of transaction inputs, since the prevout slice is built within a loop.

This PR aims to make this clear to any user who wants to reuse the logic from the example code.

### CODE REVIEW

**Comment by apoelstra** [comment]
I agree that this makes it clearer that the inputs and outputs are both arrays, and that the array of UTXOs is derived from the input array. I'm not sure it's as obvious that they're the same length as you want it to be, but it's definitely an improvement.

File: bitcoin/examples/taproot-psbt.rs [hunk 1]
```diff
@@ -267,6 +267,11 @@ fn generate_bip86_key_spend_tx(
     input.tap_internal_key = Some(input_pubkey);
     psbt.inputs = vec![input];
 
+    let mut input_txouts = Vec::<TxOut>::new();
+    for input in [&input_utxo] {
+        input_txouts.push(TxOut { value: input.amount_in_sats, script_pubkey: ScriptBuf::from_hex(input.script_pubkey)? });
+    }
```
Comment (tcharding) [inline bitcoin/examples/taproot-psbt.rs:273]: Perhaps we should add a comment on the top of `input_txouts`, something like
```
    // The `Prevouts::All` array is used to sign each input in the `psbt.inputs` array, as such it
    // must be the same length and in the same order as the inputs.
```

**Review by tcharding — APPROVED** [review]
ACK 9fe1a387549651195e787a480a10f7f0f4beda14

**Comment by tcharding** [comment]
You have a clippy warning.

**Review by apoelstra — APPROVED** [review]
ACK 215accfe09f7b835223d8b9301cc1b5636479d14

**Review by tcharding — DISMISSED** [review]
ACK 215accfe09f7b835223d8b9301cc1b5636479d14

**Comment by tcharding** [comment]
Now you have a formitting issue and a MSRV issue :) All combined you need
```rust
    for input in [&input_utxo].iter() {
        input_txouts.push(TxOut {
            value: input.amount_in_sats,
            script_pubkey: ScriptBuf::from_hex(input.script_pubkey)?,
        });
    }
```

**Review by tcharding — DISMISSED** [review]
ACK 30a4fc6a330ff0a3aa59eb1e37d4f9ac85cf45dd

**Comment by apoelstra** [comment]
Lol now clippy is complaining about the loop. You can just add a `#[allow(clippy::whatever)]` to the function, though please comment that you're doing a single-item loop deliberately to illustrate that you need to iterate over all the inputs.

**Comment by tcharding** [comment]
Gentle bump, this is one of the last PRs to go in before release, if you'd like it to.

**Comment by apoelstra** [comment]
Lol @ssantos21 I did not mean the literal text `clippy::whatever` (though I do wish that would work...) I meant to add whatever `#[allow(clippy::*)]` lint suggested by the compiler error in CI.

**Comment by ssantos21** [comment]
:laughing: 
I'm running `cargo clippy` and `cargo clippy -- -Dwarnings` but I'm not getting these warning messages.
I added `#[allow(clippy::single-element-loop)]` this time.

**Comment by tcharding** [comment]
You need to tell clippy to lint the examples as well, I typically use the following incantation `cargo clippy --all-targets --all-features` to lint.

**Comment by tcharding** [comment]
The attribute you want is `#[allow(clippy::single_element_loop)]`.

**Comment by ssantos21** [comment]
@tcharding thanks for the explanation..
Now I was able to reproduce the clippy error and fix it.
I think this new push will work.

**Review by tcharding — APPROVED** [review]
ACK 3b60ad55678518f80528b9bd313cccba43e74982

**Review by apoelstra — APPROVED** [review]
ACK 3b60ad55678518f80528b9bd313cccba43e74982

**Pull request merged**